### DISCUSSION TITLE
Proposal #2 for a new intern project

### DISCUSSION THREAD

**Discussion by ghost** [BlockchainCommons/Community#47] [Ideas]
![image](https://user-images.githubusercontent.com/13405205/123185044-ae642d00-d4b2-11eb-9f0f-786f0bdbda04.png)

## What?

[OfflineTx](https://github.com/prayank23/offlinetx): Brodcast bitcoin transactions without internet (using text messages)

## Why?

Internet [shutdowns](https://internetshutdowns.in/) have increased in India since last few years. It affects few regions regularly and text messages continue to work during these times. While most of the important things are done online today and people's life is impacted severely during shutdowns, broadcasting an important bitcoin transaction should be possible in case of an emergency.

![image](https://user-images.githubusercontent.com/13405205/123184929-6d6c1880-d4b2-11eb-8e8c-6a052180bca7.png)

Other alternatives that exist but require buying some hardware:

1. Mesh Networks
   Examples: [TxTenna](https://txtenna.com/) and [Locha](https://locha.io/)
2. Blockstream Satelite: https://blockstream.com/satellite/

## How?

1. App
2. Web Server
3. Third party APIs

**Mobile App** can be used to convert transaction hex in to words and send using text messages. Encoding of hex to words is done using [HextoWords](https://github.com/prayank23/HextoWords) which is C# implementation of [pgp-words-improved.py](https://gist.githubusercontent.com/grubles/69762512cd4e954722c03a0df1bc44ac/raw/b1a4d85dd70bfc972a6ab05a93d3432bf300d1b6/pgp-words-improved.py). I can use [URs](https://github.com/Foundation-Devices/foundation-ur-py) instead as [suggested](https://github.com/prayank23/OfflineTx/issues/1#issue-924280799) by Christopher.

**Web Server** listens for incoming text messages, saves them in a google sheet (phone number, content, flag etc.) or sql database. Second part of web server is not implemented yet which will regularly check for new data in database, decode from words to hex, broadcast using [blockstream.info](https://blockstream.info) API, save the transaction id and respond by sending text message

**APIs**: Text messaging - [Textlocal](https://api.textlocal.in/docs) Broadcast tx: [Blockstream](https://github.com/Blockstream/esplora/blob/master/API.md)

**Comment by ChristopherA** [comment]
Check with @wolfmcnally as he co-authored a paper about offline and remote use of cryptography at RWOT, and @Fonta1n3 might also be a source as when he started using bitcoin he was doing it while working at offshore oil platforms with very spotty or high latency internet connections.

**Reply by nochiel** [reply]
> [...] a paper about offline and remote use of cryptography at RWOT

Is this the paper to which you're referring: <https://github.com/WebOfTrustInfo/rwot7-toronto/blob/master/final-documents/offline-use-cases.md> ?

Aside: How much work has been done on offline off-chain transactions to recreate the way peers use physical fiat? E.g. Alice and Bob signing some kind of L3 contract to exchange value over bluetooth or using QR codes in an air-gapped mode with the same level of assurances that L2 provides? (A software version of the [OpenDime](https://opendime.com/faq) ?)

**Reply by ChristopherA** [reply]
Yes, that is the paper.

I know that there have been some discussions about the extreme offline use cases for LN, but I can’t seem to find them. I’d ask on a lightning-dev IRC channel, bitcoin-stackexchange, or even the lightning-dev list if you have a well defined question.

**Reply by ghost** [reply]
I asked a question on SE: https://bitcoin.stackexchange.com/questions/107265/offline-ln-transaction/
